src/deploy/functions/deploymentPlanner.ts

```diff
@@ -0,0 +1,175 @@
+import * as deploymentTool from "../../deploymentTool";
+import { functionMatchesAnyGroup, getTopicName } from "../../functionsDeployHelper";
+
+export interface CloudFunctionTrigger {
+  name: string;
+  sourceUploadUrl?: string;
+  labels: { [key: string]: string };
+  environmentVariables: { [key: string]: string };
+  entryPoint: string;
+  runtime?: string;
+  vpcConnector?: string;
+  vpcConnectorEgressSettings?: string;
+  ingressSettings?: string;
+  availableMemoryMb?: number;
+  timeout?: number;
+  maxInstances?: number;
+  serviceAccountEmail?: string;
+  httpsTrigger?: any;
+  eventTrigger?: any;
+  failurePolicy?: {};
+  schedule?: object;
+  timeZone?: string;
+  regions?: string[];
+}
+
+export interface RegionMap {
+  [region: string]: CloudFunctionTrigger[];
+}
+
+export interface RegionalDeployment {
+  region: string;
+  sourceToken?: string;
+  firstFunctionDeployment?: CloudFunctionTrigger;
+  functionsToCreate: CloudFunctionTrigger[];
+  functionsToUpdate: CloudFunctionTrigger[];
+  schedulesToCreateOrUpdate: CloudFunctionTrigger[];
+}
+
+export interface DeploymentPlan {
+  regionalDeployments: RegionalDeployment[];
+  functionsToDelete: string[];
```

Comment:
Scheduler jobs are technically regional - however, they can only live in the default resource location (aka the region where app engine exists on that project). Because of this, they are effectively global.

FunctionsToDelete is also regional - however, we don't need to wait for the first function in that region to be deployed to  fire off the delete calls, so I figured it would be simpler to put them all in one array.